### ISSUE TITLE
How to find the pre-trained model of p6？

### ISSUE THREAD

**Issue by autumn2332** [ultralytics/ultralytics#24693]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

In the context of yolo26, yolov8, and other yolo models, how can I obtain the p6 pre-trained model for these models? I intend to train a target detection model based on this p6 model. I look forward to your prompt reply

**Comment by UltralyticsAssistant** [comment]
👋 Hello @autumn2332, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you started, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
For your specific question, it would be helpful if you could clarify which model family and exact checkpoint naming or configuration you are looking for, along with your current Ultralytics package version and any steps you have already tried 🔍
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/get-started/locally/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by raimbekovm** [comment]
ready-made pretrained p6 weights only exist for yolov5. these download automatically by name:

```python
from ultralytics import YOLO
model = YOLO("yolov5x6u.pt")  # also n6/s6/m6/l6
model.train(data="your.yaml", imgsz=1280, epochs=100)
```

for yolo26 and yolov8 there are p6 model configs but no official pretrained p6 checkpoints, so you point at the yaml and train from scratch:

```python
model = YOLO("yolo26-p6.yaml")  # prefix the scale: yolo26n-p6.yaml ... yolo26x-p6.yaml
model.train(data="your.yaml", imgsz=1280, epochs=300)
```

if you want a head start, you can warm-start the backbone from the standard p5 checkpoint, though only the early backbone carries over (the p6 neck and head reindex), so it's a partial transfer rather than a real pretrained p6:

```python
model = YOLO("yolo26-p6.yaml").load("yolo26x.pt")  # match the scale
```

the configs are listed in the [model config guide](https://docs.ultralytics.com/usage/cfg/), and the [yolov5 page](https://docs.ultralytics.com/models/yolov5/) covers the pretrained p6 lineup.

**Comment by onuralpszr** [comment]
As @raimbekovm mentation we don't have yolo26 p6 pre-trained so I am closing issue as complete and thank you to all 👍 🚀

**Comment by autumn2332** [comment]
First of all, we would like to extend our sincere gratitude to your entire team for the outstanding work you've done on the YOLO series. Thank you so much again for all the support you've provided for my development work, with which I am able to train a high-quality model that perfectly meets my practical needs.

---Original---
From: "Murat ***@***.***&gt;
Date: Thu, Jun 4, 2026 14:19 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ultralytics/ultralytics] How to find the pre-trained model of p6？ (Issue #24693)


raimbekovm left a comment (ultralytics/ultralytics#24693)
 
ready-made pretrained p6 weights only exist for yolov5. these download automatically by name:
 from ultralytics import YOLO model = YOLO("yolov5x6u.pt")  # also n6/s6/m6/l6 model.train(data="your.yaml", imgsz=1280, epochs=100)
 
for yolo26 and yolov8 there are p6 model configs but no official pretrained p6 checkpoints, so you point at the yaml and train from scratch:
 model = YOLO("yolo26-p6.yaml")  # prefix the scale: yolo26n-p6.yaml ... yolo26x-p6.yaml model.train(data="your.yaml", imgsz=1280, epochs=300)
 
if you want a head start, you can warm-start the backbone from the standard p5 checkpoint, though only the early backbone carries over (the p6 neck and head reindex), so it's a partial transfer rather than a real pretrained p6:
 model = YOLO("yolo26-p6.yaml").load("yolo26x.pt")  # match the scale
 
the configs are listed in the model config guide, and the yolov5 page covers the pretrained p6 lineup.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by glenn-jocher** [comment]
Thanks for the kind words — the credit really goes to the YOLO community and the Ultralytics team, and I’m glad the guidance helped you get a model working well for your use case.
